lib/Provider/MailService.php

```diff
@@ -0,0 +1,202 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+namespace OCA\Mail\Provider;
+
+use OCP\Mail\Provider\Address;
+use OCP\Mail\Provider\IAddress;
+use OCP\Mail\Provider\IMessage;
+use OCP\Mail\Provider\IMessageSend;
+use OCP\Mail\Provider\IService;
+use OCP\Mail\Provider\Message;
+
+use Psr\Container\ContainerInterface;
+
+class MailService implements IService, IMessageSend {
+
+	protected array $serviceSecondaryAddresses = [];
+	protected array $serviceAbilities = [];
+
+	public function __construct(
+		protected ContainerInterface $container,
+		protected string $userId = '',
+		protected string $serviceId = '',
+		protected string $serviceLabel = '',
+		protected IAddress $servicePrimaryAddress = new Address(),
+	) {
+		$this->serviceAbilities = ['MessageSend' => true];
+	}
+
+	/**
+	 * arbitrary unique text string identifying this service
+	 *
+	 * @since 4.0.0
+	 *
+	 * @return string id of this service (e.g. 1 or service1 or anything else)
+	 */
+	public function id(): string {
+
+		return $this->serviceId;
+
```

Comment:
```suggestion
		return $this->serviceId;
```

same for the rest of the diff